store/src/java/com/zimbra/cs/service/mail/ModifyCalendarItem.java

```diff
@@ -108,8 +109,9 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         MailSendQueue sendQueue = new MailSendQueue();
         Element response = getResponseElement(zsc);
         int compNum = (int) request.getAttributeLong(MailConstants.A_CAL_COMP, 0);
-        mbox.lock.lock();
+        final MailboxLock l = mbox.lock(true);
```

Comment:
This is correct. It does not change control flow. An alternative to this is what I put above https://github.com/Zimbra/zm-mailbox/pull/444/files#r154731075